AwarenessApisSampleKotlin/app/src/main/java/com/android/example/awarenessapissamplekotlin/MainActivity.kt

```diff
@@ -37,6 +55,83 @@ class MainActivity : AppCompatActivity() {
 
     fun onClickSnapshot(view: View) {
         Log.d(TAG, "onClickSnapshot()")
+
+        if (activityRecognitionPermissionApproved()) {
+            requestSnapshot()
+        } else {
+            requestPermissionLauncher.launch(Manifest.permission.ACTIVITY_RECOGNITION)
+        }
+    }
+
+    private fun requestSnapshot() {
+        Log.d(TAG, "requestSnapshot()")
+
+        val task: Task<DetectedActivityResponse> =
```

Comment:
is it possible to simplify this to: `val task = Awareness.getSnapshotClient(this).detectedActivity`?

Reply:
Yes, done. :) I included the explicit type for devs reading online, but it's implied clearly in callbacks, so you are right, I can remove.